src/intrprtr.c

```diff
@@ -657,14 +657,14 @@ void            IntrIfEnd (
 **  Since loops cannot be interpreted immediately,  the interpreter calls the
 **  coder  to create a  procedure (with no arguments) and  calls that.
 */
-void IntrForBegin ( void )
+void IntrForBegin(Obj stackNams)
 {
     /* ignore                                                              */
     SKIP_IF_RETURNING();
     SKIP_IF_IGNORING();
 
     if (STATE(IntrCoding) == 0)
-        StartFakeFuncExpr(0);
+        StartFakeFuncExpr(stackNams, 0);
```

Comment:
Note that `StartFakeFuncExpr` previously accessed the ReaderState, to be able to access `StackNams`. But since `StartFakeFuncExpr ` is called from the interpreter, which doesn't have access to the ReaderState (and I'd like to keep it that way), we can't just pass the ReaderState to `StartFakeFuncExpr`. The solution then is to just pass `StackNams` to a few select interpreter functions.

All in all, I am not really satisfied with this solution; but then, I've grappled with this very problem (which boils down to the interpreter calling back to the reader, which otherwise doesn't happen) for quite some time (1-2 years?) and couldn't come up with a satisfactory solution so far. The only other approach (besides the one I used here) which I didn't hate completely was to add return values to `IntrForBegin` and friends to indicate to the reader that it should call  `StartFakeFuncExpr`. resp. `FinishAndCallFakeFuncExpr`. But this relies on `CodeForBegin` etc. to not do anything (which is true, and probably *safe* to rely on, but still not "nice"; but OK, I could live with that). In any case, it felt to me that it just made it even harder to reason about what's going on... 

That said if somebody has an idea how to do this "better" / "differently", I am eager to hear, so please don't hesitate to speak up!